From 604b68fabf535ee8eaf6e381e1fe82d03d14d290 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Tue, 30 Oct 2018 15:02:10 +0800
Subject: [PATCH 4953/5242] MLK-20124-1: soc: imx: keep hdmi tx power domains
 always on

commit  065b17975a4fbd4f9c04091aed42481f4bf881d3 from
https://source.codeaurora.org/external/imx/linux-imx.git

HDMI TX should keep in power on state in suspend mode.
Otherwise HDMI FW will lost.
Add flag GENPD_FLAG_ALWAYS_ON to hdmi tx power domains.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index 9a636dad..3c7f725 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -90,6 +90,14 @@ static int imx8_pd_power(struct generic_pm_domain *domain, bool power_on)
 		return -EINVAL;
 	}
 
+	/* keep HDMI TX resource power on */
+	if (power_on && (pd->rsrc_id == SC_R_HDMI ||
+					pd->rsrc_id == SC_R_HDMI_I2S ||
+					pd->rsrc_id == SC_R_HDMI_I2C_0 ||
+					pd->rsrc_id == SC_R_HDMI_PLL_0 ||
+					pd->rsrc_id == SC_R_HDMI_PLL_1))
+		pd->pd.flags |= GENPD_FLAG_ALWAYS_ON;
+
 	return 0;
 }
 
-- 
1.7.9.5

